src/PrestaShopBundle/Controller/Admin/Improve/Payment/PaymentPreferencesController.php

```diff
@@ -52,21 +53,47 @@ class PaymentPreferencesController extends FrameworkBundleAdminController
      */
     public function indexAction(Request $request)
     {
+        $separationEnabled = $this->get('prestashop.core.admin.feature_flag.repository')->isEnabled(FeatureFlagSettings::FEATURE_FLAG_PAYMENT_PREFERENCES_FORM_SEPARATION);
         $legacyController = $request->attributes->get('_legacy_controller');
 
         $paymentModulesListProvider = $this->get('prestashop.adapter.module.payment_module_provider');
         $shopContext = $this->get('prestashop.adapter.shop.context');
 
         $isSingleShopContext = $shopContext->isSingleShopContext();
 
-        $paymentPreferencesForm = null;
         $paymentModulesCount = 0;
-
         if ($isSingleShopContext) {
             $paymentModulesCount = count($paymentModulesListProvider->getPaymentModuleList());
+        }
+
+        if ($separationEnabled) {
+            $carrierRestrictionsView = $countryRestrictionsView = $currencyRestrictionsView = $groupRestrictionsView = null;
+            if ($isSingleShopContext) {
+                $carrierRestrictionsView = $this->getPaymentCarrierRestrictionsFormHandler()->getForm()->createView();
+                $countryRestrictionsView = $this->getPaymentCountryRestrictionsFormHandler()->getForm()->createView();
+                $currencyRestrictionsView = $this->getPaymentCurrencyRestrictionsFormHandler()->getForm()->createView();
+                $groupRestrictionsView = $this->getPaymentGroupRestrictionsFormHandler()->getForm()->createView();
+                $paymentModulesCount = count($paymentModulesListProvider->getPaymentModuleList());
+            }
+
+            return $this->render('@PrestaShop/Admin/Improve/Payment/Preferences/separated_payment_preferences.html.twig', [
+                'enableSidebar' => true,
+                'help_link' => $this->generateSidebarLink($legacyController),
+                'paymentCurrencyRestrictionsForm' => $currencyRestrictionsView,
+                'paymentCountryRestrictionsForm' => $countryRestrictionsView,
+                'paymentGroupRestrictionsForm' => $groupRestrictionsView,
+                'paymentCarrierRestrictionsForm' => $carrierRestrictionsView,
+                'isSingleShopContext' => $isSingleShopContext,
+                'paymentModulesCount' => $paymentModulesCount,
+            ]);
+        }
+
+        $paymentPreferencesForm = null;
+        if ($isSingleShopContext) {
             $paymentPreferencesForm = $this->getPaymentPreferencesFormHandler()->getForm()->createView();
         }
 
+        /* In next major version for separate views should be passed one for each restriction group */
```

Comment:
sorry to pick up on a comment, but I don't understand what it means. maybe the "for" is not needed, is that what you meant?
```suggestion
        /* In next major version separate views should be passed one for each restriction group */
```
By the way with "should" it sounds like a todo instead of simple comment? But mentioning next major version sounds kind of like a deprecation notice? Which is it? (just want to make sure the purpose of this comment is clear and it will not get outdated and useless :smile: )

Reply:
It's a todo notice, moved a comment to where it's relevant and marked as todo